File: app/src/main/java/nz/org/cacophony/birdmonitor/views/FrequencyFragment.java
package nz.org.cacophony.birdmonitor.views;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Switch;
import android.widget.TextView;
import nz.org.cacophony.birdmonitor.Prefs;
import nz.org.cacophony.birdmonitor.R;
import nz.org.cacophony.birdmonitor.Util;

public class FrequencyFragment extends Fragment {

    private static final String TAG = "FrequencyFragment";


    private Switch swUseFrequentUploads;
    private TextView tvMessages;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_frequency, container, false);
        setUserVisibleHint(false);

        tvMessages = view.findViewById(R.id.tvMessages);
        swUseFrequentUploads = view.findViewById(R.id.swUseFrequentUploads);
        displayOrHideGUIObjects();

        swUseFrequentUploads.setOnCheckedChangeListener((buttonView, isChecked) -> {
            Util.setUseFrequentUploads(getActivity().getApplicationContext(), swUseFrequentUploads.isChecked());
            displayOrHideGUIObjects();
        });

        return view;
    }

    @Override
    public void setUserVisibleHint(final boolean visible) {
        super.setUserVisibleHint(visible);
        if (getActivity() == null) {
            return;
        }
        if (visible) {
            displayOrHideGUIObjects();
        }
    }


    void displayOrHideGUIObjects() {
        Prefs prefs = new Prefs(getActivity());
        swUseFrequentUploads.setChecked(prefs.getUseFrequentUploads());

        if (prefs.getUseFrequentUploads()) {
            swUseFrequentUploads.setText("Upload after every recording is ON");
        } else {
            swUseFrequentUploads.setText("Upload after every recording is OFF");
        }

    }

}

File: app/src/main/java/nz/org/cacophony/birdmonitor/views/FrequencyFragment.kt
package nz.org.cacophony.birdmonitor.views

import nz.org.cacophony.birdmonitor.Util.setUseFrequentUploads
import android.widget.TextView
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.CompoundButton
import android.widget.Switch
import androidx.core.widget.doAfterTextChanged
import androidx.core.widget.doOnTextChanged
import androidx.fragment.app.Fragment
import nz.org.cacophony.birdmonitor.Prefs
import nz.org.cacophony.birdmonitor.R
import nz.org.cacophony.birdmonitor.Util
import kotlin.random.Random
import kotlin.random.nextLong

class FrequencyFragment : Fragment() {
    private var swUseFrequentUploads: Switch? = null
    private var tvMessages: TextView? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        val view = inflater.inflate(R.layout.fragment_frequency, container, false)
        userVisibleHint = false
        tvMessages = view.findViewById(R.id.tvMessages)
        swUseFrequentUploads = view.findViewById(R.id.swUseFrequentUploads)
        displayOrHideGUIObjects()
        swUseFrequentUploads?.setOnCheckedChangeListener(CompoundButton.OnCheckedChangeListener { _: CompoundButton?, isChecked: Boolean ->
            setUseFrequentUploads(
                requireActivity().applicationContext, isChecked
            )
            displayOrHideGUIObjects()
        })
        val prefs = Prefs(requireActivity().applicationContext)
        val randomText = view.findViewById<TextView>(R.id.RandomSeedText)
        val randomButton = view.findViewById<Button>(R.id.RandomSeedButton)
        val randomSeed = prefs.randomSeed.toString()
        randomText.text = randomSeed
        // add a listener to the button
        randomButton.setOnClickListener {
            val newSeed = Random.nextLong(9999)
            randomText.text = newSeed.toString()
            prefs.randomSeed = newSeed
            val alarm = Util.getNextAlarm(requireActivity().applicationContext, prefs, null)
            prefs.setTheNextSingleStandardAlarmUsingUnixTime(alarm.TimeMillis)
        }
        
        randomText.doAfterTextChanged { text ->
            if (text.isNullOrEmpty()) {
                return@doAfterTextChanged
            }
            val newSeed = text.toString().toLong()
            prefs.randomSeed = newSeed
            val alarm = Util.getNextAlarm(requireActivity().applicationContext, prefs, null)
            prefs.setTheNextSingleStandardAlarmUsingUnixTime(alarm.TimeMillis)
        }

        return view
    }

    override fun setUserVisibleHint(visible: Boolean) {
        super.setUserVisibleHint(visible)
        if (activity == null) {
            return
        }
        if (visible) {
            displayOrHideGUIObjects()
        }
    }

    fun displayOrHideGUIObjects() {
        val prefs = Prefs(activity)
        swUseFrequentUploads!!.isChecked = prefs.useFrequentUploads
        if (prefs.useFrequentUploads) {
            swUseFrequentUploads!!.text = "Upload after every recording is ON"
        } else {
            swUseFrequentUploads!!.text = "Upload after every recording is OFF"
        }
    }

    companion object {
        private const val TAG = "FrequencyFragment"
    }
}